v-next/hardhat/test/internal/builtin-plugins/network-manager/hook-handlers/config.ts

```diff
@@ -604,6 +604,399 @@ describe("network-manager/hook-handlers/config", () => {
         );
       });
     });
+
+    describe("accounts", () => {
+      describe("http config", async () => {
+        let hardhatUserConfig: any; // Use any to allow assigning also wrong values
+
+        before(() => {
+          hardhatUserConfig = {
+            networks: {
+              localhost: {
+                type: "http",
+                accounts: "", // Modified in the tests
+                url: "http://localhost:8545",
+              },
+            },
+          };
+        });
+
+        describe("allowed values", () => {
+          it("should allow the value 'remote'", async () => {
+            hardhatUserConfig.networks.localhost.accounts = "remote";
+
+            const validationErrors =
+              await validateUserConfig(hardhatUserConfig);
+
+            console.log(validationErrors);
+
+            assert.equal(validationErrors.length, 0);
+          });
+
+          it("should allow an array of valid private keys", async () => {
+            hardhatUserConfig.networks.localhost.accounts = [
+              "0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+              "0xbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb",
+              "0xcccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc",
+            ];
+
+            const validationErrors =
+              await validateUserConfig(hardhatUserConfig);
+
+            assert.equal(validationErrors.length, 0);
+          });
+
+          it("should allow an account with a valid HttpNetworkHDAccountsConfig", async () => {
+            hardhatUserConfig.networks.localhost.accounts = {
+              mnemonic: "asd asd asd",
+              initialIndex: 0,
+              count: 123,
+              path: "m/123",
+              passphrase: "passphrase",
+            };
+
+            const validationErrors =
+              await validateUserConfig(hardhatUserConfig);
+
+            assert.equal(validationErrors.length, 0);
+          });
+
+          // TODO: still valid?
```

Comment:
Do we still wanna keep the possibility of allowing valid private keys with missing hex prefix?
If yes, I will have to modify the hh-utils method that I added to check if an address is valid, [see here](https://github.com/NomicFoundation/hardhat/pull/5817/files#diff-8d9a1ba6601820ff6249438ad3797d59980861eef0ca459e2a05512362894bcd)